File: solutions/15SortColors/java/SortColors.java
/**
 * SortColors
 */
public class SortColors {

  public void sortColors(int[] nums) {
    int n = nums.length;
    int lastRed = 0, firstBlue = n - 1, index = 0;
    while (index <= firstBlue) {
      if (index < lastRed) {
        index++;

      } else if (nums[index] == 0) {
        int tmp = nums[index];
        nums[index] = nums[lastRed];
        nums[lastRed] = tmp;
        lastRed++;
      } else if (nums[index] == 2) {

        int tmp = nums[index];
        nums[index] = nums[firstBlue];
        nums[firstBlue] = tmp;
        firstBlue--;
      } else {
        index++;
      }
    }
  }

}

File: solutions/15SortColors/python/SortColors.py
class Solution(object):
    def sortColors(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        lastRed, firstBlue, index = 0, n - 1, 0
        while index <= firstBlue:
            if index < lastRed:
                index += 1
            elif nums[index] == 0:
                nums[index], nums[lastRed] = nums[lastRed], nums[index]
                lastRed += 1
            elif nums[index] == 2:
                nums[index], nums[firstBlue] = nums[firstBlue], nums[index]
                firstBlue -= 1
            else:
                index += 1
